fancy_collections/formatting.py

```diff
@@ -45,25 +47,49 @@ def key_to_string(self, key: Any, obj: Any) -> str:  # noqa
     def to_string(self) -> str:
         if len(self._obj) == 0:
             return self._stringify_empty_class(self._obj)
+
+        objects = {}
+        widths = {}
         for key, val in self._obj.items():
             key = self.key_to_string(key, val)
-            try:
-                string = self.stringify(val)
-            except Exception:  # noqa
-                string = NotImplemented
-            if string is NotImplemented:
-                string = str(val)
-            self._add(key, string.splitlines())
+            lines = self.stringify(val).splitlines()
+            objects[key] = lines
+            widths[key] = self._get_maxlen(lines + [key])
+
+        display_width = int(shutil.get_terminal_size().columns * 0.8)
+
+        keys = tuple(widths.keys())
+        front_keys, back_keys = set(), set()
+        line_length = 0
+        for fkey, bkey in zip(keys, keys[::-1]):
+            line_length += widths[fkey]
+            if line_length < display_width:
+                front_keys.add(fkey)
+            line_length += widths[bkey]
+            if line_length < display_width:
+                back_keys.add(bkey)
+
+        for k, v in objects.items():
+            if k in front_keys:
+                self._add(k, v)
+
+        if not front_keys.intersection(back_keys):
+            self._add("...", ["..."])
```

Comment:
consider columns `a,b,c,d` and `front_keys={a,b}` and `back_keys={c,d}`, and line length is exhausted. 
We would create a header `"a | b | ... | c | d"` even though there are no more columns in the `"..."`, or do i miss something ?

Reply:
You are right... The following should work, right?

```suggestion
        if not front_keys.intersection(back_keys) and front_keys.union(back_keys) != set(keys):
            self._add("...", ["..."])
```